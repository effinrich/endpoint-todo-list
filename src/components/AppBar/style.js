import styled from 'styled-components'
import { Flex } from 'reflexbox'

export default styled(Flex)`
  top: 0;
  left: 0;
  position: fixed;
  width: 100%;
  z-index: 1;
  min-height: 72px;
  background-color: white;
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.2);
`
