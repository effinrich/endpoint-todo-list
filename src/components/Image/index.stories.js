import React from 'react'

import Image from '.'

export default {
  title: 'UI Library/Image',
  component: Image
}

const Template = (args) => <Image {...args} />

export const Default = Template.bind({})
Default.args = {
  src:
    'https://s-media-cache-ak0.pinimg.com/originals/6d/96/41/6d9641691af89520f3b19b26bca4b5fc.jpg',
  alt: 'kitty'
}

export const Thumb = Template.bind({})
Thumb.args = {
  src:
    'https://s-media-cache-ak0.pinimg.com/originals/6d/96/41/6d9641691af89520f3b19b26bca4b5fc.jpg',
  alt: 'kitty',
  thumb: true,
  maxWidth: 300
}

export const Border = Template.bind({})
Border.args = {
  src:
    'https://s-media-cache-ak0.pinimg.com/originals/6d/96/41/6d9641691af89520f3b19b26bca4b5fc.jpg',
  alt: 'kitty',
  border: true
}

export const WMissingImage = Template.bind({})
WMissingImage.args = {
  src: 'brokenimageurl.com',
  alt: 'kitty'
}

export const WAltImagePropAndBrokenImage = Template.bind({})
WAltImagePropAndBrokenImage.args = {
  src:
    'https://s-media-cache-ak0.pinimg.com/originals/6d/96/41/6d9641691af89520f3b19b26bca4b5fc.jpg',
  alt: 'kitty',
  altImage: true
}
