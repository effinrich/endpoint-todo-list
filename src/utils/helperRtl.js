import React from 'react'
import PropTypes from 'prop-types'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'

import store from 'store'

const Wrapper = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

Wrapper.propTypes = {
  children: PropTypes.object
}

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
