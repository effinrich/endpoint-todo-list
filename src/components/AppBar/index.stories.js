import React from 'react'
import { Box } from 'reflexbox'

import Image from 'components/Image'
import Logo from 'assets/logo.svg'

import AppBar from '.'

export default {
  title: 'UI Library/AppBar',
  component: AppBar,
  parameters: {
    componentSubtitle:
      'Displays a fixed app bar header for branding, navigation, etc.'
  }
}

const Template = args => (
  <AppBar {...args}>
    <Box>
      <Image maxWidth={240} alt="logo" src={Logo} />
    </Box>
  </AppBar>
)

export const Default = Template.bind({})
