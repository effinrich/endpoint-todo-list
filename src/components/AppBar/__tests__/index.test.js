import React from 'react'
import { Box } from 'reflexbox'

import { render } from 'utils/helperRtl'
import Image from 'components/Image'
import Logo from 'assets/logo.svg'

import AppBar from '..'

const children = () => (
  <Box>
    <Image maxWidth={240} alt="logo" src={Logo} />
  </Box>
)

describe('AppBar Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<AppBar />)
    expect(container).toMatchSnapshot()
  })

  it('should render a logo image', () => {
    const { getByAltText } = render(<AppBar children={children()} />)
    expect(getByAltText(/logo/i)).toBeInTheDocument()
  })
})
