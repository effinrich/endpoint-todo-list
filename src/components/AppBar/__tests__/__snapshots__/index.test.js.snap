// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component should render without crashing 1`] = `
<div>
  <div
    class="sc-gsTCUz gMIZDD css-k9o90o"
  />
</div>
`;
